plugins/out_doris/doris_conf.c

```diff
@@ -0,0 +1,241 @@
+#include <windows.h>
+#endif
+
+static inline void atomic_store(int *dest, int val) {
+#ifdef FLB_SYSTEM_WINDOWS
+    InterlockedExchange((LONG volatile *) dest, (LONG) val);
+#else
+    __atomic_store_n(dest, val, __ATOMIC_RELEASE);
+#endif
+}
+
+static inline int atomic_load(int *dest) {
+#ifdef FLB_SYSTEM_WINDOWS
+    return (int) InterlockedCompareExchange((LONG volatile *) dest, 0, 0);
+#else
+    return __atomic_load_n(dest, __ATOMIC_ACQUIRE);
+#endif
+}
+
+void *report(void *c) {
+    struct flb_out_doris *ctx = (struct flb_out_doris *) c;
+    
+    size_t init_time = cfl_time_now() / 1000000000L;
+    size_t last_time = init_time;
+    size_t last_bytes = ctx->reporter->total_bytes;
+    size_t last_rows = ctx->reporter->total_rows;
+
+    size_t cur_time, cur_bytes, cur_rows, total_time, total_speed_mbps, total_speed_rps;
+    size_t inc_bytes, inc_rows, inc_time, inc_speed_mbps, inc_speed_rps;
+
+    flb_plg_info(ctx->ins, "Start progress reporter with interval %d", ctx->log_progress_interval);
+    
+    while (atomic_load(&ctx->reporter->running) && ctx->log_progress_interval > 0) {
+        flb_time_msleep(ctx->log_progress_interval * 1000);
+
+        cur_time = cfl_time_now() / 1000000000L;
+        cur_bytes =  ctx->reporter->total_bytes;
+        cur_rows =  ctx->reporter->total_rows;
+        total_time = cur_time - init_time;
+        total_speed_mbps = cur_bytes / 1024 / 1024 / total_time;
+        total_speed_rps = cur_rows / total_time;
+
+        inc_bytes = cur_bytes - last_bytes;
+		inc_rows = cur_rows - last_rows;
+		inc_time = cur_time - last_time;
+		inc_speed_mbps = inc_bytes / 1024 / 1024 / inc_time;
+		inc_speed_rps = inc_rows / inc_time;
+
+        flb_plg_info(ctx->ins, "total %zu MB %zu ROWS, total speed %zu MB/s %zu R/s, last %zu seconds speed %zu MB/s %zu R/s",
+			         cur_bytes/1024/1024, cur_rows, total_speed_mbps, total_speed_rps,
+			         inc_time, inc_speed_mbps, inc_speed_rps);
+        
+        last_time = cur_time;
+		last_bytes = cur_bytes;
+		last_rows = cur_rows;
+    }
+
+    return NULL;
+}
+
+struct flb_out_doris *flb_doris_conf_create(struct flb_output_instance *ins,
+                                            struct flb_config *config)
+{
+    int ret;
+    int io_flags = 0;
+    const char *tmp;
+    struct flb_upstream *upstream;
+    struct flb_out_doris *ctx = NULL;
+    struct flb_doris_progress_reporter *reporter = NULL;
+
+    /* Allocate plugin context */
+    ctx = flb_calloc(1, sizeof(struct flb_out_doris));
+    if (!ctx) {
+        flb_errno();
+        return NULL;
+    }
+    ctx->ins = ins;
+
+    ret = flb_output_config_map_set(ins, (void *) ctx);
+    if (ret == -1) {
+        flb_free(ctx);
+        return NULL;
+    }
+
+    /* Set default network configuration */
+    flb_output_net_default("127.0.0.1", 8030, ins);
+
+    /* Validate */ 
+    if (!ctx->endpoint_type || (strcasecmp(ctx->endpoint_type, "fe") != 0 && strcasecmp(ctx->endpoint_type, "be") != 0)) {
```

Comment:
Please simplify this construct so it's easiert to read. Perform a direct comparison against `NULL` and separate the string comparisons in a new line so the code reads in a way that's conducive to a quick and correct interpretation of the motive.